src/main/java/com/github/wokier/maven/progress/ProgressUpdater.java

```diff
@@ -0,0 +1,156 @@
+
+package com.github.wokier.maven.progress;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+import java.util.Date;
+
+import org.apache.maven.plugin.MojoExecutionException;
+
+/**
+ * Updates the progress of a reactor by storing and retrieving the progress from a properties file.
+ * Because the plugin can run multithreaded in several classloaders a file is used to record state.
```

Comment:
I am building with Maven 3.x with the -T 1C option. 

Here is a link that describes when Maven uses several ClassLoaders: http://takari.io/book/91-maven-classloading.html

So looks like it depends on the plugins you use if you are using them with different configurations around your projects.

I encountered it while building some projects with:
* flexmojos-maven-plugin and <packaging>swf</packaging> in the POM
* android-maven-plugin and <packaging>apk</packaging> in the POM